app/javascript/elm/src/SessionsList.elm

```diff
@@ -0,0 +1,192 @@
+port module SessionsList exposing (Model, Msg(..), Session, ShortType, checkedSession, init, loadMoreSessions, main, subscriptions, update, updateSessions, view, viewSession, viewShortType)
+
+import Browser
+import Html exposing (Html, button, dd, div, dl, dt, h1, img, input, label, li, section, span, text, ul)
+import Html.Attributes exposing (checked, class, for, id, src, style, type_)
+import Html.Events exposing (onClick)
+import String exposing (fromInt)
+
+
+
+---- MODEL ----
+
+
+type alias ShortType =
+    { name : String
+    , type_ : String
+    }
+
+
+type alias Session =
+    { title : String
+    , id : Int
+    , timeframe : String
+    , username : String
+    , shortTypes : List ShortType
+    , selected : Bool -- THIS IS USED IN JS, IN ELM WE TRACK SELECTION IN selectedSessionId. REMOVE WHEN MOVING FETCHING FROM JS TO ELM
+    }
+
+
+type alias Model =
+    { sessions : List Session
+    , selectedSessionId : Maybe Int
+    }
+
+
+init : List Session -> ( Model, Cmd Msg )
+init flags =
+    ( { sessions = flags, selectedSessionId = Nothing }, Cmd.none )
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.batch [ updateSessions UpdateSessions ]
+
+
+
+---- UPDATE ----
+
+
+port updateSessions : (List Session -> msg) -> Sub msg
+
+
+port checkedSession : { deselected : Maybe Int, selected : Maybe Int } -> Cmd msg
+
+
+port loadMoreSessions : () -> Cmd msg
+
+
+type Msg
+    = ToggleSessionSelection Int
+    | UpdateSessions (List Session)
+    | LoadMoreSessions
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        ToggleSessionSelection id ->
+            if model.selectedSessionId == Just id then
+                ( { model | selectedSessionId = Nothing }
+                , checkedSession { deselected = model.selectedSessionId, selected = Nothing }
+                )
+
+            else
+                ( { model | selectedSessionId = Just id }
+                , checkedSession { deselected = model.selectedSessionId, selected = Just id }
+                )
+
+        UpdateSessions sessions ->
+            let
+                selectedSessionId =
+                    List.head << List.map .id << List.filter .selected
+            in
+            ( { model | sessions = sessions, selectedSessionId = selectedSessionId sessions }, Cmd.none )
+
+        LoadMoreSessions ->
+            ( model, loadMoreSessions () )
+
+
+
+---- VIEW ----
+
+
+viewSession : Maybe Int -> Session -> Html Msg
+viewSession selectedSessionId session =
+    li
+        [ style "padding" "5px 10px"
```

Comment:
What do you think about keeping styling in some css file and just using classes here?

Reply:
We are gonna remove that when moving to the new design so I didn't want to overthink the css.